### ISSUE TITLE
More on Formatter documentation

### ISSUE THREAD

**Issue by Manuelbaun** [eclipse-langium/langium#1221]
I am trying to get the formatter working as in the domain-model example. 
I use the exact same setup as described in the docs : https://langium.org/guides/formatting/

Now running the Extension in VSCode I try to format my DSL. When call the `format document` command, the format function of my formatter does not get called. It only gets called, when I call the `format selection` command from VSCode. 

I can't seem to figure out how I can run the formatter for the whole document. 

I would like to have some more documentation in order to get the formatter work properly.  In the showcase example https://langium.org/showcase/domainmodel/, it seems to work.

**Comment by msujew** [comment]
Weird. The `textDocument/formatting` and `textDocument/rangeFormatting` LSP requests go basically through the same function. See [doDocumentFormat](https://github.com/eclipse-langium/langium/blob/a4c7a58e9a79199d0f69cd3a9479c6ddfe99de21/packages/langium/src/lsp/formatter.ts#L131-L148). Can you share a reproducible example?

**Comment by Manuelbaun** [comment]
You are right (sorry for the noise). I thought I did setup something wrong. While setting up a reproducible example, I used the `domain-model` example and it worked as expected. Now I found now why:

I tried to implement a terminal rule, which wasn't correctly parsed yet, I had a syntax error. This caused the formatter not to execute. 

I guess, It would be good, to actually display a message, that the formatter cannot be executed or something.
